crates/biome_js_analyze/src/lint/nursery/no_noninteractive_element_interactions.rs

```diff
@@ -0,0 +1,302 @@
+    "onDblClick",
+    "onDoubleClick",
+    "onDrag",
+    "onDragEnd",
+    "onDragEnter",
+    "onDragExit",
+    "onDragLeave",
+    "onDragOver",
+    "onDragStart",
+    "onDrop",
+    "onMouseDown",
+    "onMouseEnter",
+    "onMouseLeave",
+    "onMouseMove",
+    "onMouseOut",
+    "onMouseOver",
+    "onKeyDown",
+    "onKeyPress",
+    "onKeyUp",
+    "onFocus",
+    "onBlur",
+    "onLoad",
+    "onError",
+];
+
+type AttributesRef<'a> = Option<&'a FxHashMap<String, Vec<String>>>;
+
+/// Check if the element contains event handler
+fn has_handler_props(attributes: AttributesRef) -> bool {
+    attributes.is_some_and(|m| {
+        INTERACTIVE_HANDLERS
+            .iter()
+            .any(|handler| m.contains_key(*handler))
+    })
+}
+
+/// Check if the element's implicit ARIA semantics have been removed.
+/// See https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/presentation_role
+///
+/// Ref: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/v6.10.0/src/util/isPresentationRole.js
+fn has_presentation_role(attributes: AttributesRef) -> bool {
+    const PRESENTATION_ROLE: &[&str] = &["presentation", "none"];
+
+    if let Some(attributes) = attributes {
+        if let Some(values) = attributes.get("role") {
+            let is_presentation = values
+                .iter()
+                .any(|v| PRESENTATION_ROLE.contains(&v.as_str()));
+            return is_presentation;
+        }
+    }
+    false
+}
+
+/// Check the element is hidden from screen reader.
+/// See
+/// - https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-hidden
+/// - https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/hidden
+///
+/// Ref: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/v6.10.0/src/util/isHiddenFromScreenReader.js
+fn is_hidden_from_screen_reader(element_name: &str, attributes: AttributesRef) -> bool {
+    if let Some(attributes) = attributes {
+        let aria_hidden = attributes
+            .get("aria-hidden")
+            .map_or(false, |attr| attr.contains(&"true".to_string()));
+
+        let input_hidden = element_name == "input"
+            && attributes
+                .get("type")
+                .map_or(false, |attr| attr.contains(&"hidden".to_string()));
+
+        return aria_hidden || input_hidden;
+    }
+
+    false
+}
+
+/// Check if the element is `contentEditable`
+/// See https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/contenteditable
+///
+/// Ref: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/blob/v6.10.0/src/util/isContentEditable.js
+fn is_content_editable(attributes: AttributesRef) -> bool {
+    if let Some(attributes) = attributes {
+        if let Some(values) = attributes.get("contentEditable") {
+            return values.contains(&"true".to_string());
+        }
+    }
+    false
+}
+
+/// List of `widget` role
+/// https://www.w3.org/TR/wai-aria-1.2/#widget
+const INTERACTIVE_ROLES: &[&str] = &[
```

Comment:
This should probably be coming from our `biome_aria` crate. Preferably generated from the spec.

Reply:
Thank you for your review!
I’ve updated the code to use `biome_aria` as suggested (https://github.com/biomejs/biome/pull/4358/commits/df9cbe01ad84a8410e19fc43fa9bc15acea9b0a4). 
However, I noticed a difference in the test results after this change. 
For example, the `radiogroup` role, which inherits from `select` and should be interactive, is now being treated as non-interactive.

```bash
         54 │+# Diagnostics
         55 │+```
         56 │+valid.jsx:49:5 lint/nursery/noNoninteractiveElementInteractions ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
         57 │+
         58 │+  ! Non-interactive element should not have event handler.
         59 │+  
         60 │+  > 49 │     <div role="radiogroup" onClick={() => { }} />
         61 │+       │     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
         62 │+    50 │ </>
         63 │+  
         64 │+  i Consider replace semantically interactive element like <button/> or <a href/>.
         65 │+  
         66 │+
 ```
 
 According to WAI-ARIA 1.1 and 1.2, radiogroup inherits from `select`, but role.rs indicates inheritance from `range`.
 https://github.com/biomejs/biome/blob/19abdcf49ec67ad53e9d9fb36b8e74fe60b4c603/crates/biome_aria/src/roles.rs#L487-L493
 
https://www.w3.org/TR/wai-aria-1.1/#radiogroup
https://www.w3.org/TR/wai-aria-1.2/#radiogroup

I am not familiar with accessibility, so I apologize if there is any misunderstanding.
Could you please advise me on how to proceed?

Reply:
That's probably a bug, right? You could try to add the missing roles and see if that fixes the issue?

Reply:
I manually added the `widget` role and the `radiogroup` is no longer reported.

```diff
--- a/crates/biome_aria/src/roles.rs
+++ b/crates/biome_aria/src/roles.rs
@@ -488,7 +488,7 @@ define_role! {
     /// https://www.w3.org/TR/wai-aria-1.1/#radiogroup
     RadiogroupRole {
         PROPS: [("aria-readonly", false), ("aria-required", false)],
-        ROLES: ["range"],
+        ROLES: ["range","widget"],
     }
 }
 ```
 
 ```bash
Expression: valid.jsx
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
-old snapshot
+new results
────────────┬──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
   44    44 │     <div role="textbox" onClick={() => { }} />
   45    45 │ 
   46    46 │     {/* Presentation role */}
   47    47 │     <div role="presentation" onClick={() => { }} />
         48 │+
         49 │+
         50 │+    <div role="radiogroup" onClick={() => { }} />
   48    51 │ </>
   49    52 │ ```
   ```
   
   If it's not my implementation error, I think it's a bug.